fix flickering on map resize

pstaszek opened the pull request:
This PR fixes issue #4158.

When screen size changes, we immediately resize the canvas. Resizing canvas causes canvas to clear. Then, next render is scheduled using `_update` method. Because it is scheduled, and it happens asynchronously, in the time between the canvas is white.

There are two ways of solving the problem. Simple - in this PR, is to trigger render every time when canvas changes.
Other option would be to resize the canvas asynchronosly as well. I didn't try this approach, as it seems more complex, and I'm not sure if it is worth.

The downside of current approach is that the render might be called multiple time, which might impact performance. Although I'm not sure if this is an issue.

In this PR I replaced `_update` with `redraw` which happens immediately. Also `_update` also updates style and sources, which doesn't seem for me like something that should happen during resizing. Can you see any reason to do that?

In this PR https://github.com/maplibre/maplibre-gl-js/pull/4429 `_render` method was added, which is not ideal, because already frame was scheduled, and also triggering `_render` directly is not clearing scheduled frames, like `redraw`.


 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Add an entry to `CHANGELOG.md` under the `## main` section.

HarelM commented:
Looks good, thanks for the info! 
Want chance you can add a test to this somehow?

pstaszek commented:
@HarelM
Yes! I updated unit test, and checked if `redraw` and `_render` are called after `resize`. I think this is pretty good check, as it fails if `_render` was called asynchronously (for example if I change for `_update`, spy on `_render` was not called).

I also added comment on unit test.

HarelM reviewed (approved):
THANKS!
